### ISSUE TITLE
Eval bug: >9116 llama-server crash using draft model

### ISSUE THREAD

**Issue by Black6spdZ** [ggml-org/llama.cpp#23023]
### Name and Version

C:\test>llama-server --version
load_backend: loaded RPC backend from C:\test\ggml-rpc.dll
load_backend: loaded Vulkan backend from C:\test\ggml-vulkan.dll
load_backend: loaded CPU backend from C:\test\ggml-cpu-haswell.dll
version: 9118 (706fbd8ab)
built with Clang 19.1.5 for Windows x86_64

### Operating systems

Windows

### GGML backends

Vulkan

### Hardware

i7-6850X, 64GB quad-channel ddr4-2400, Radeon Pro v620 32GB

### Models

Qwen3.6-27b main, Qwen3.5-0.8b draft

### Problem description & steps to reproduce

get a crash as soon as the model is prompted. 9116 works as intended

### First Bad Commit

https://github.com/ggml-org/llama.cpp/pull/22693

### Relevant log output

srv  update_slots: all slots are idle
srv  params_from_: Chat format: peg-native
slot get_availabl: id  0 | task -1 | selected slot by LRU, t_last = -1
srv  get_availabl: updating prompt cache
srv          load:  - looking for better prompt, base f_keep = -1.000, sim = 0.000
srv        update:  - cache state: 0 prompts, 0.000 MiB (limits: 8192.000 MiB, 131072 tokens, 8589934592 est)
srv  get_availabl: prompt cache update took 0.02 ms
reasoning-budget: activated, budget=2147483647 tokens
slot launch_slot_: id  0 | task -1 | sampler chain: logits -> ?penalties -> ?dry -> ?top-n-sigma -> top-k -> ?typical -> top-p -> min-p -> ?xtc -> ?temp-ext -> dist
slot launch_slot_: id  0 | task 0 | processing task, is_child = 0
slot update_slots: id  0 | task 0 | new prompt, n_ctx_slot = 131072, n_keep = 0, task.n_tokens = 1380
slot update_slots: id  0 | task 0 | n_tokens = 0, memory_seq_rm [0, end)
srv  log_server_r: done request: POST /v1/chat/completions 10.1.1.180 200
slot update_slots: id  0 | task 0 | prompt processing progress, n_tokens = 512, batch.n_tokens = 512, progress = 0.371015

C:\test>

**Comment by 0cc4m** [comment]
should already be fixed, check again.

**Comment by Black6spdZ** [comment]
all good! just retested with latest 9172
